assets/js/components/KeyMetrics/ChangeMetricsLink.js

```diff
@@ -48,7 +50,46 @@ export default function ChangeMetricsLink() {
 		trackEvent( `${ viewContext }_kmw`, 'change_metrics' );
 	}, [ setValue, viewContext ] );
 
-	if ( ! Array.isArray( keyMetrics ) || ! keyMetrics?.length > 0 ) {
+	const renderChangeMetricLink =
+		Array.isArray( keyMetrics ) || keyMetrics?.length > 0;
+
+	const isKeyMetricsSetupCompleted = useSelect( ( select ) =>
+		select( CORE_SITE ).isKeyMetricsSetupCompleted()
+	);
+
+	const keyMetricsSetupCompletedByUserID = useSelect( ( select ) =>
+		select( CORE_SITE ).getKeyMetricsSetupCompletedByUserID()
+	);
+
+	const currentUserID = useSelect( ( select ) =>
+		select( CORE_USER ).getID()
+	);
+
+	const { triggerOnDemandTour } = useDispatch( CORE_USER );
+
+	useEffect( () => {
+		const isUserEligibleForTour =
+			Number.isInteger( keyMetricsSetupCompletedByUserID ) &&
+			Number.isInteger( currentUserID ) &&
+			keyMetricsSetupCompletedByUserID > 0 &&
+			currentUserID !== keyMetricsSetupCompletedByUserID;
+
+		if (
+			renderChangeMetricLink &&
+			isKeyMetricsSetupCompleted &&
+			isUserEligibleForTour
+		) {
```

Comment:
There's a fair amount of code here which is only needed for this effect rather than the component itself. How about we extract it to a custom hook, e.g. `useChangeMetricsFeatureTourEffect`? Not strictly necessary of course, but would be a nice refactoring if time allows.